src/cli/java/org/commcare/util/cli/ApplicationHost.java

```diff
@@ -319,15 +320,20 @@ private boolean loopSession() throws IOException {
                     // which ultimately updates the session, so getNextScreen will move onto the form list,
                     // skipping the entity detail. To avoid this, flag that we want to force a redraw in this case.
                     boolean waitForCaseDetail = false;
-                    if (screen instanceof EntityScreen) {
-                        boolean isAction = input.startsWith("action "); // Don't wait for case detail if action
-                        EntityScreen eScreen = (EntityScreen)screen;
-                        if (!isAction && eScreen.getCurrentScreen() instanceof EntityListSubscreen) {
-                            waitForCaseDetail = true;
+                    if (screen instanceof MultiSelectEntityScreen) {
+                        String[] selectedValues = input.split(",");
+                        screen.handleInputAndUpdateSession(mSession, input, false, selectedValues);

```

Comment:
This should pass in the `USE_SELECTED_VALUES` const along with the list of values instead of the raw input.
```suggestion
                        screen.handleInputAndUpdateSession(mSession, USE_SELECTED_VALUES, false, selectedValues);

```